File: test-plugins/org.yakindu.sct.generator.cpp.test/gtests/TypeAliasTest/TypeAliasTest.cc
#include <string>
#include "gtest/gtest.h"
#include "TypeAlias.h"
#include "sc_runner.h"
#include "sc_types.h"

namespace  {

TypeAlias* statechart;



//! The timers are managed by a timer service. */
static SctUnitRunner * runner;

class TypeAliasTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		statechart = new TypeAlias();
		statechart->init();
		runner = new SctUnitRunner(
			statechart,
			false,
			200
		);
	}
	virtual void TearDown() {
		delete statechart;
		delete runner;
	}
};


TEST_F(TypeAliasTest, TypeAliasTest) {
	
	statechart->enter();
	
	EXPECT_TRUE(statechart->isStateActive(TypeAlias::main_region_Start));
	
	EXPECT_TRUE(statechart->getDefaultSCI()->get_myVar()== 1l);
	
	runner->proceed_cycles(1);
	
	EXPECT_TRUE(statechart->isStateActive(TypeAlias::main_region_Mid));
	
	EXPECT_TRUE(strcmp(statechart->getDefaultSCI()->get_myString(), "TypeSystem") == 0);
	
	runner->proceed_cycles(1);
	
	EXPECT_TRUE(statechart->isStateActive(TypeAlias::main_region_Mid2));
	
	statechart->getDefaultSCI()->raise_myEvent();
	
	runner->proceed_cycles(1);
	
	EXPECT_TRUE(statechart->isStateActive(TypeAlias::main_region_End));
	
	
}

	}

File: test-plugins/org.yakindu.sct.generator.java.test/test-gen/org/yakindu/sct/generator/java/test/TypeAliasTest.java
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/

package org.yakindu.sct.generator.java.test;

import org.junit.*;
import static org.junit.Assert.*;
import org.yakindu.scr.typealias.TypeAliasStatemachine;
import org.yakindu.scr.typealias.TypeAliasStatemachine.State;

/**
 * Unit TestCase for TypeAlias
 */
@SuppressWarnings("all")
public class TypeAliasTest {
	private TypeAliasStatemachine statemachine;	
	
	@Before
	public void setUp() {
		statemachine = new TypeAliasStatemachine();
		statemachine.init();
	}

	@After
	public void tearDown() {
		statemachine = null;
	}
	
	@Test
	public void testTypeAliasTest() {
		statemachine.enter();
		assertTrue(statemachine.isStateActive(State.main_region_Start));
		assertTrue(statemachine.getMyVar() == 1l);
		statemachine.runCycle();
		assertTrue(statemachine.isStateActive(State.main_region_Mid));
		assertTrue(statemachine.getMyString().equals("TypeSystem"));
		statemachine.runCycle();
		assertTrue(statemachine.isStateActive(State.main_region_Mid2));
		statemachine.raiseMyEvent();
		statemachine.runCycle();
		assertTrue(statemachine.isStateActive(State.main_region_End));
	}
}
